### PR TITLE
Wpcli_all_the_way

### PR DESCRIPTION
**Pull request by pioneerskies**
This is going to be merged into the 6.0 version branch.

This will bring a big feature: standing on the shoulder of the 6.0's mandatory peer dependency on wp-cli, we going to drop all the custom methods and configuration for the `local` environment relative to

- db connection
- db dump
- db import
- db adapt
- wordmove site URL (former `local.vhost` config in movefile.yml)

All these actions and config will be done/get through wp-cli.

Now the only configuration we'll have for local environment will be

```yml
local:
  wordpress_path: /some/path
```

The goals of this feature are:

- simplifying configuration
- rely as much as possible on what wp-cli already can do

`doctor` command will still check for database config (user, password, etc.), erroring when actually we can't get those configs from wp-cli.

`port` and `mysqldump_options` and `mysql_options` and `charset` database configs are dropped: if you need to tweak around those, you'll need to do it the wordpress way (https://wordpress.org/support/article/editing-wp-config-php) and the wp-cli way (https://make.wordpress.org/cli/handbook/references/config/#config-files).

Last but not the least: this branch will update ruby language support to the current stable version 3.0.2.

Nonetheless the vision steering this feature was endorsed in #601 . I'm not taking all the ideas from @nlemoine , but I'm looking at what I can afford to bring out of alpha within my winter holidays ^^"

### CODE REVIEW

**Comment by pioneerskies** [comment]
I've also updated the `movefile.yml` generator called by `wordmove init` command to reflect current changes

File: lib/wordmove/movefile.rb [hunk 1]
```diff
@@ -75,18 +73,40 @@ def fetch(verbose = true) # rubocop:disable Style/OptionalBooleanParameter
...
+
+      options.merge(
+        local: {
+          database: {
+            password: Wordmove::WpcliHelpers.get_config('DB_PASSWORD', config_path: config_path),
+            host: Wordmove::WpcliHelpers.get_config('DB_HOST', config_path: config_path),
+            name: Wordmove::WpcliHelpers.get_config('DB_NAME', config_path: config_path),
+            user: Wordmove::WpcliHelpers.get_config('DB_USER', config_path: config_path)
+          },
+          vhost: Wordmove::WpcliHelpers.get_option('siteurl', config_path: config_path),
```
Comment (nlemoine) [inline lib/wordmove/movefile.rb:96]: I think It should be `home` not `siteurl` which in a "Bedrock" like structure will return something like `https://www.example.com/wp`

**Comment by nlemoine** [comment]
This is awesome! Thanks so much for implementing all these new features 😃  I just left a tiny comment.

File: lib/wordmove/movefile.rb (hunk 1 above)
Comment (pioneerskies) [reply lib/wordmove/movefile.rb:96]: Woah. This is a terrific catch!

Going to fix it in #624 since this is already merged :)

**Pull request merged**